src/python/pants/backend/scala/target_types.py

```diff
@@ -36,9 +37,12 @@ class ScalaSettingsRequest(TargetFilesGeneratorSettingsRequest):
 
 
 @rule
-def scala_settings_request(_: ScalaSettingsRequest) -> TargetFilesGeneratorSettings:
-    # TODO: See https://github.com/pantsbuild/pants/issues/14382.
-    return TargetFilesGeneratorSettings(add_dependencies_on_all_siblings=True)
+def scala_settings_request(
+    scala_infer_subsystem: ScalaInferSubsystem, _: ScalaSettingsRequest
+) -> TargetFilesGeneratorSettings:
+    return TargetFilesGeneratorSettings(
+        add_dependencies_on_all_siblings=scala_infer_subsystem.force_add_siblings_as_dependencies
```

Comment:
I think this should probably be:

`scala_infer_subsystem.force_add_siblings_as_dependencies or not scala_infer_subsystem.imports`